flumine/markets/market.py

```diff
@@ -62,6 +62,13 @@ def close_market(self) -> None:
             extra=self.info,
         )
 
+    def belongs_to_strategy(self, strategy) -> bool:
+        """
+        Returns True if the market belongs to the given strategy, else False.
+        """
+        # self.market_book is always defined since Market object is initialised with it
+        return self.market_book.streaming_unique_id in strategy.stream_ids
```

Comment:
That's a very good point. It is not a problem for `_process_market_books` because we don't want to call a strategy which is not subscribed to that stream, but it is a problem for `_process_market_catalogues` because the stream id is used to back-reference the strategy which needs to be called.

Instead of checking for stream ids, we could validate the market against the strategy's market filter and see if it is compatible with it or not. It would be a slower check than testing for stream id membership, but it should not matter because market catalogue updates occur infrequently.

Reply:
Resolved in the latest commit. A method is added which checks if a market filter matches the market. It is a more comprehensive check, but only invoked if testing by stream id fails.